### PR TITLE
[Test] Add Rocky8 to tests.

### PR DESCRIPTION
**Pull request by gmarciani**
### Description of changes
Add Rocky8 to kitchen and spec tests.

### Tests
* Verified that ` ./kitchen.ec2.sh platform-install test nvidia-gdrcopy` launches test for Rocky8.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-environment/spec/unit/resources/cloudwatch_spec.rb [hunk 1]
```diff
@@ -33,7 +33,16 @@ def self.configure(chef_run)
       cached(:signature_path) { "#{package_path}.sig" }
 
       context "when not on arm" do
-        cached(:platform_url_component) { platform == 'amazon' ? 'amazon_linux' : platform }
+        cached(:platform_url_component) do
+          case platform
+          when 'amazon'
+            'amazon_linux'
+          when 'rocky'
```
Comment (jdeamicis) [inline cookbooks/aws-parallelcluster-environment/spec/unit/resources/cloudwatch_spec.rb:40]: Any reason why CentOS x86_64 differs from CentOS arm (see your next change at line 117)?

File: cookbooks/aws-parallelcluster-platform/spec/unit/resources/install_packages_spec.rb [hunk 2]
```diff
@@ -53,6 +55,22 @@ def self.setup(chef_run)
...
+        end
+
+        it 'installs kernel source' do
+          is_expected.to run_bash('Install kernel source')
+            .with(user: 'root')
+            .with_code(/set -e/)
+            .with_code(/dnf install -y #{kernel_source_package}-#{kernel_source_package_version} --releasever #{platform}/)
+            .with_code(/dnf clean all/)
+        end
+
```
Comment (jdeamicis) [inline cookbooks/aws-parallelcluster-platform/spec/unit/resources/install_packages_spec.rb]: I would move this up and do the following:
```ruby
if %w(amazon centos redhat rocky).include?(platform)
  it 'installs default packages' do
    # This is common even to Rocky 8
    ...
  end
  
  if platform == 'rocky'
    it 'installs kernel source' do
       # Rocky methodology via bash block
       ...
    end
  else
    it 'install kernel source' do
      # Other RHEL derivatives methodology via `package`
      ...
    end
  end
end
```

**Review by jdeamicis — DISMISSED** [review]
Approved with two comments.

File: cookbooks/aws-parallelcluster-environment/spec/unit/resources/cloudwatch_spec.rb (hunk 1 above)
Comment (gmarciani) [reply cookbooks/aws-parallelcluster-environment/spec/unit/resources/cloudwatch_spec.rb:40]: There is no CWagent for CentOS on ARM, so we fall back to RHEL package.
This is not the case for CentOS x86.

File: cookbooks/aws-parallelcluster-platform/spec/unit/resources/install_packages_spec.rb (hunk 2 above)
Comment (gmarciani) [reply cookbooks/aws-parallelcluster-platform/spec/unit/resources/install_packages_spec.rb]: Done

**Review by jdeamicis — APPROVED** [review]

**Pull request merged**